### PR TITLE
[android] Yet another LocationHelper fix fix.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mgsergio or @bykoianko or @yunikkk or @trashkalmar PTAL

https://fabric.io/mapsme/android/apps/com.mapswithme.maps.pro/issues/57b33942ffcdc04250c0aec0

### CODE REVIEW

File: android/src/com/mapswithme/maps/location/LocationHelper.java [hunk 1]
```diff
@@ -75,10 +76,21 @@ public void onTransit(boolean foreground)
         final IntentFilter filter = new IntentFilter();
         filter.addAction(LocationManager.PROVIDERS_CHANGED_ACTION);
         filter.addCategory(Intent.CATEGORY_DEFAULT);
-        MwmApplication.get().registerReceiver(mReceiver, filter);
+
+        if (!mReceiverRegistered)
+        {
```
Comment (trashkalmar) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Its better to move `IntentFilter` object initialization inside this block.

File: android/src/com/mapswithme/maps/location/LocationHelper.java [hunk 2]
```diff
@@ -63,10 +63,11 @@
     boolean shouldNotifyLocationNotFound();
   }
 
-  private final GPSCheck mReceiver = new GPSCheck();
-
   private final OnTransitionListener mOnTransition = new OnTransitionListener()
   {
+    private final GPSCheck mReceiver = new GPSCheck();
+    private boolean mReceiverRegistered = false;
```
Comment (trashkalmar) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Field is set to `false` by default. No explicit initialization is required.

**Comment by ygorshenin** [comment]
PTAL

File: android/src/com/mapswithme/maps/location/LocationHelper.java (hunk 1 above)
Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Done

File: android/src/com/mapswithme/maps/location/LocationHelper.java (hunk 2 above)
Comment (ygorshenin) [inline android/src/com/mapswithme/maps/location/LocationHelper.java]: Done

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by bykoianko** [comment]
LGTM

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by mapsmetest** [comment]
FormatChecker

**Comment by ygorshenin** [comment]
Thanks!

**Comment by ygorshenin** [comment]
FormatChecker

**Comment by ygorshenin** [comment]
FormatChecker FormatChecker FormatChecker

**Comment by yunikkk** [comment]
LGTM

**Pull request merged**